walletunlocker/service.go

```diff
@@ -313,6 +342,9 @@ func (u *UnlockerService) InitWallet(ctx context.Context,
 func (u *UnlockerService) UnlockWallet(ctx context.Context,
 	in *lnrpc.UnlockWalletRequest) (*lnrpc.UnlockWalletResponse, error) {
 
+	u.Lock()
+	defer u.Unlock()
```

Comment:
I think we may also need another signal/bool at the `UnlockerService` level that indicates the wallet has been initialized/unlocked so that we can check it here and everywhere else, otherwise it seems like we still risk attempting to open the wallet twice.

Reply:
What do you mean? Push things down even further? With where things are atm, we won't _return_ back to the caller until the wallet has been fully initialized. However, for unlock we return a bit earlier once we have all the credentials. In my testing, the concurrent init was was ended up tripping things up. 

For unlock, things only become "fully finalized" once we create the chain control.

Reply:
Actually, as soon as this method returns, unlocking isn't even possible since the listener service only has a lifetime of this call.